### PR TITLE
ARIA-42-Generic-ctx-serialization-mechanism

### PR DESCRIPTION
**Pull request by mxmrlv**

### CODE REVIEW

File: aria/storage/sql_mapi.py [hunk 1]
```diff
@@ -15,13 +15,18 @@
 """
 SQLAlchemy based MAPI
 """
+import os
+import platform
 
+from sqlalchemy import create_engine
```
Comment (mxmrlv) [inline aria/storage/sql_mapi.py]: fix imports

File: aria/storage/sql_mapi.py [hunk 2]
```diff
@@ -364,6 +369,28 @@ def _load_relationships(instance):
...
+    if base_dir is not None:
+        uri = 'sqlite:///{platform_char}{path}'.format(
+            # Handles the windows behavior where there is not root, but drivers.
+            # Thus behaving as relative path.
+            platform_char='' if 'Windows' in platform.system() else '/',
+
+            path=os.path.join(base_dir, filename))
+        engine_kwargs = {}
+    else:
+        uri = 'sqlite:///:memory:'
```
Comment (mxmrlv) [inline aria/storage/sql_mapi.py]: remove in memory option to tests only

File: tests/mock/context.py [hunk 3]
```diff
@@ -15,26 +15,30 @@
...
 
-def simple(mapi_kwargs, resources_dir=None, **kwargs):
-    model_storage = aria.application_model_storage(SQLAlchemyModelAPI, api_kwargs=mapi_kwargs)
+def simple(tmpdir,
+           model_initiator_kwargs=None,
+           resources_initiator_kwargs=None,
+           context_kwargs=None):
 
-    deployment_id = create_simple_topology_two_nodes(model_storage)
+    model_storage = aria.application_model_storage(
```
Comment (mxmrlv) [inline tests/mock/context.py]: be consistent with initiator passing

File: tests/orchestrator/workflows/executor/test_executor.py [hunk 4]
```diff
@@ -80,9 +81,9 @@ class MockException(Exception):
     pass
 
 
-class MockContext(object):
+class MockContext(operation.BaseOperationContext):
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs):                                                            # pylint: disable=super-init-not-called
```
Comment (mxmrlv) [inline tests/orchestrator/workflows/executor/test_executor.py]: add serialize function

File: tests/orchestrator/workflows/executor/test_process_executor.py [hunk 5]
```diff
@@ -106,9 +112,9 @@ def mock_plugin(plugin_manager, tmpdir):
     return plugin_manager.install(source=plugin_path)
 
 
-class MockContext(object):
+class MockContext(operation.BaseOperationContext):
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs):                                                            # pylint: disable=super-init-not-called
```
Comment (mxmrlv) [inline tests/orchestrator/workflows/executor/test_process_executor.py]: ditto

File: aria/storage/core.py [hunk 6]
```diff
@@ -66,10 +67,10 @@ def __init__(self,
         self.registered = {}
         self._initiator = initiator
         self._initiator_kwargs = initiator_kwargs or {}
+        self._api_kwargs = api_kwargs or {}
+        self._additional_api_kwargs = {}
```
Comment (ran-z) [inline aria/storage/core.py]: add information about initiator interface

Comment (ran-z) [inline aria/storage/core.py]: unpicklable

File: tests/storage/__init__.py [hunk 7]
```diff
@@ -65,3 +65,14 @@ def release_sqlite_storage(storage):
             session.close()
         for engine in set(mapi._engine for mapi in mapis):
```
Comment (ran-z) [inline tests/storage/__init__.py]: dont forget to revisit mock context

**Pull request merged**